### PR TITLE
Add next-mysql example

### PR DESCRIPTION
**Pull request by msweeneydev**
This PR adds a Next.js and MySQL example using the `serverless-mysql` module. It also covers the importance of using the module as well as an explanation of how serverless architecture can work with MySQL.

### CODE REVIEW

File: nextjs-mysql/.nowignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules
+.env
+.gitignore
```
Comment (styfle) [inline nextjs-mysql/.nowignore]: You can remove `.nowignore`. The default ignored files are:

```
.hg
.git
.gitmodules
.svn
.npmignore
.dockerignore
.gitignore
.*.swp
.DS_Store
.wafpicke-*
.lock-wscript
.env
.env.build
npm-debug.log
config.gypi
node_modules
__pycache__
CVS
```

File: nextjs-mysql/README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+Lastly, the `next.config.js` is used to tell [Next.js](https://nextjs.org) that we are targeting a serverless environment.
+
+## How Serverless MySQL Works
+
+Serverless functions create multiple database connections for each user and can use up all available connections quickly, blocking access for others. This example resolves this issue by utilizing [Jeremy Daly's](http://www.jeremydaly.com) MySQL wrapper - [serverless-mysql](https://github.com/jeremydaly/serverless-mysql).
+
+The wrapper provides basic but important functionality that allows connections to be monitored, limited, retried and closed. This is vital for making serverless MySQL work and is explored further in `db.js` below:
+
+```js
+const mysql = require("serverless-mysql");
```
Comment (styfle) [inline nextjs-mysql/README.md]: Change your double quotes to single quotes

File: nextjs-mysql/README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+By using `await db.end()`, `serverless-mysql` ensures the connection is closed before sending the response to the client.
+
+As a result, it ensures that concurrent MySQL connections do not reach their limit, allowing new connections to be created when required.
+
+Next, let's look at `/api/profiles/profile.js` to see what a serverless MySQL query looks like:
+
+```js
+const db = require("../../lib/db");
+const sql = require("sql-template-strings");
+const url = require("url");
```
Comment (styfle) [inline nextjs-mysql/README.md]: Change your double quotes to single quotes

File: nextjs-mysql/api/profiles/index.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const db = require("../../lib/db");
+const sql = require("sql-template-strings");
+const url = require("url");
```
Comment (styfle) [inline nextjs-mysql/api/profiles/index.js]: All the other examples use single quotes.

Lets change the double quotes to single quotes.

File: nextjs-mysql/README.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
+# Next.js + MySQL Example
+
+This directory is an example of a [Next.js](https://nextjs.org) app connected to a [MySQL](https://www.mysql.com/) database using the [@now/next](https://zeit.co/docs/v2/deployments/official-builders/next-js-now-next/) and [@now/node](https://zeit.co/docs/v2/deployments/official-builders/node-js-now-node/) builders.
```
Comment (styfle) [inline nextjs-mysql/README.md]:
```suggestion
This directory is an example of a [Next.js](https://nextjs.org) app connected to a [MySQL](https://www.mysql.com/) database using the [@now/next](https://zeit.co/docs/v2/deployments/official-builders/next-js-now-next/) and [@now/node](https://zeit.co/docs/v2/deployments/official-builders/node-js-now-node/) Builders.
```

Comment (styfle) [reply nextjs-mysql/README.md]: We should capitalize Builders

File: nextjs-mysql/.nowignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+node_modules
```
Comment (styfle) [inline nextjs-mysql/.nowignore]: Actually, maybe you want to ignore the `db.sql` file. Is that used at runtime?

Comment (msweeneydev) [reply nextjs-mysql/.nowignore]: Good catch, no it's not.

File: nextjs-mysql/README.md (hunk 5 above)
Comment (msweeneydev) [reply nextjs-mysql/README.md]: Changed and added link to docs.

File: nextjs-mysql/README.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+The example includes a `now.json` file, used to configure your build when [deploying](https://zeit.co/docs/v2/deployments/basics/) to Now, and a `.nowignore` file, used to [prevent uploading unnecessary source paths](https://zeit.co/guides/prevent-uploading-sourcepaths-with-nowignore/) to Now, keeping deployment times short.
+
+Also included is a `db.sql` file, which has the example data and SQL commands, this can be used to replicate this example.
+
+Lastly, the `next.config.js` is used to tell [Next.js](https://nextjs.org) that we are targeting a serverless environment.
+
+## How Serverless MySQL Works
+
+Serverless functions create multiple database connections for each user and can use up all available connections quickly, blocking access for others. This example resolves this issue by utilizing [Jeremy Daly's](http://www.jeremydaly.com) MySQL wrapper - [serverless-mysql](https://github.com/jeremydaly/serverless-mysql).
```
Comment (styfle) [inline nextjs-mysql/README.md]:
```suggestion
Serverless functions will create multiple database connections as traffic increases and therefore all available connections can be consumed quickly, blocking access for others. This example resolves this issue by utilizing [Jeremy Daly's](http://www.jeremydaly.com) MySQL wrapper - [serverless-mysql](https://github.com/jeremydaly/serverless-mysql).
```

It's not true that each user gets their own connection. One user might have 2 devices or two users might not click concurrently therefore the connection is reused.

**Pull request merged**